ledger-tool/src/output.rs

```diff
@@ -327,162 +331,116 @@ impl EncodedConfirmedBlockWithEntries {
-                let tx_status_meta = blockstore
-                    .read_transaction_status((tx_signature, slot))
-                    .unwrap_or_else(|err| {
-                        eprintln!(
-                            "Failed to read transaction status for {} at slot {}: {}",
-                            transaction.signatures[0], slot, err
-                        );
-                        None
-                    })
-                    .map(|meta| meta.into());
-
-                solana_cli_output::display::println_transaction(
-                    &transaction,
-                    tx_status_meta.as_ref(),
-                    "      ",
-                    None,
-                    None,
-                );
-            }
-        }
-        OutputFormat::Json => {
-            // Note: transaction status is not output in JSON yet
-            serde_json::to_writer(stdout(), &entry).expect("serialize entry");
-            stdout().write_all(b",\n").expect("newline");
-        }
-        _ => unreachable!(),
-    }
+pub(crate) fn encode_confirmed_block(
+    confirmed_block: ConfirmedBlock,
+) -> Result<EncodedConfirmedBlock> {
+    let encoded_block = confirmed_block
+        .encode_with_options(
+            UiTransactionEncoding::Base64,
+            BlockEncodingOptions {
+                transaction_details: TransactionDetails::Full,
+                show_rewards: true,
+                max_supported_transaction_version: Some(0),
+            },
+        )
+        .map_err(|err| match err {
+            EncodeError::UnsupportedTransactionVersion(version) => LedgerToolError::Generic(
+                format!("Failed to process unsupported transaction version ({version}) in block"),
+            ),
+        })?;
+    let encoded_block: EncodedConfirmedBlock = encoded_block.into();
+    Ok(encoded_block)
 }
 
 pub fn output_slot(
     blockstore: &Blockstore,
     slot: Slot,
     allow_dead_slots: bool,
-    method: &OutputFormat,
+    output_format: &OutputFormat,
     verbose_level: u64,
     all_program_ids: &mut HashMap<Pubkey, u64>,
-) -> Result<(), String> {
+) -> Result<()> {
     if blockstore.is_dead(slot) {
         if allow_dead_slots {
-            if *method == OutputFormat::Display {
+            if *output_format == OutputFormat::Display {
                 println!(" Slot is dead");
             }
         } else {
-            return Err("Dead slot".to_string());
+            return Err(LedgerToolError::from(BlockstoreError::DeadSlot));
         }
     }
 
-    let (entries, num_shreds, is_full) = blockstore
-        .get_slot_entries_with_shred_info(slot, 0, allow_dead_slots)
-        .map_err(|err| format!("Failed to load entries for slot {slot}: {err:?}"))?;
+    let Some(meta) = blockstore.meta(slot)? else {
+        return Ok(());
+    };
+    let VersionedConfirmedBlockWithEntries { block, entries } = blockstore
+        .get_complete_block_with_entries(
```

Comment:
Use of this function now requires a full block to be present, we would previously make a best effort to display entries based on what we had available. I can see how displaying partial slots could be good, but thinking personally on this, I usually have two cases:
- See if I have the full slot
- Print the full slot

So, I have convinced myself that only printing full blocks here is ok

Reply:
We return `SlotUnavailable` for a partial slot, yeah? That's less explicit than it could be. What do you think about emitting a trace or debug log if `get_complete_block_with_entries` is called on a partial slot?

It might be nice to have some way to print a partial slot, just in case, but not sure what that UI should be. I'm fine with normal `agave-ledger-tool slot` operation printing only full slots.

Reply:
> What do you think about emitting a trace or debug log if get_complete_block_with_entries is called on a partial slot?

Works for me - I see another `info` that is arguably a bit noisy, going to demote that one to trace as well while I'm in there